src/main/java/za/ac/cput/factory/VehicleFactory.java

```diff
@@ -0,0 +1,30 @@
+/*      VehicleFactory.java
+        Factory for the Vehicle Entity
+        Author: Simphiwe Kahlana (220162891)
+        Date: 22 March 2023
+        */
+
+package za.ac.cput.factory;
+
+import za.ac.cput.domain.Branch;
+import za.ac.cput.domain.Vehicle;
+import za.ac.cput.util.Helper;
+
+public class VehicleFactory {
+    public static Vehicle createVehicle(String VehicleId, String Make, String Model, String Type, String Colour, int Year, int Mileage){
+
+        if (Helper.isNullOrEmpty(VehicleId) || Helper.isNullOrEmpty(Make) || Helper.isNullOrEmpty(Model) || Helper.isNullOrEmpty(Type)
+                || Helper.isNullOrEmpty(Colour)|| Helper.isNullOrEmpty(Year)|| Helper.isNullOrEmpty(Mileage))
+            return null;
+        //PLEASE LET ME KNOW IF THERE'S SOMETHING MISSING
+
+        return new Vehicle.Builder().setVehicleId(VehicleId)
```

Comment:
Create the vehicleId using the generateId() method provided in the helper class, that way each time a Vehicle is created, it's ensured that the id will be unique.

```suggestion
           public static Vehicle createVehicle(String Make, String Model, String Type, String Colour, int Year, int Mileage){

        if (Helper.isNullOrEmpty(Make) || Helper.isNullOrEmpty(Model) || Helper.isNullOrEmpty(Type)
                || Helper.isNullOrEmpty(Colour)|| Helper.isNullOrEmpty(Year)|| Helper.isNullOrEmpty(Mileage))
            return null;

        return new Vehicle.Builder().setVehicleId(generateId())...
```